cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/statements/Statement.kt

```diff
@@ -67,6 +70,8 @@ abstract class Statement : Node(), DeclarationHolder {
     override fun addDeclaration(declaration: Declaration) {
         if (declaration is VariableDeclaration) {
```

Comment:
can't we just add all value declarations here? do we really need to differentiate what they are?

Reply:
I have no idea. If there would be any reason to **not** support one of {`EnumConstantDeclaration`, `ParameterDeclaration`, `ProblemDeclaration`, `TypeParameterDeclaration`}, we could keep this check (but it's a bit sneaky and nobody will realize a problem). Similarly, if we would like to differentiate between different types of declarations, it could make sense. If we can't come up with anything useful, I'll just add all `ValueDeclaration`s here

Reply:
Let's keep it for now, I want to re design the addDeclaration bit anyway, probably as part of the NodeBuilder v2 API.